lib/THC/THCTensorMath.cuh

```diff
@@ -23,4 +23,31 @@ __global__ void THCTensor_copyToDiagonal(T* a, T* b, ptrdiff_t start, ptrdiff_t
   }
 }
 
+#define CAT_ARRAY_KERNEL_BATCH_SIZE 2
```

Comment:
So I am not super familiar with constant memory, but it seems to me like you do something like:

`__constant__ int deviceMem[128];`

and populate it via a call to `cudaMemcpyToSymbol(...)`. However, given that our TensorInfo structs have a template parameter, I am not sure if we can use this interface, as the size of the array would not be known at compile time.